### PR TITLE
add support to set node taints during NodeDeployment creation

### PR DESCRIPTION
**Pull request by mrIncompetent**
**What this PR does / why we need it**:
Add support to set node taints during NodeDeployment creation.

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

File: api/pkg/api/v1/types.go [hunk 1]
```diff
@@ -593,6 +600,8 @@ type NodeSpec struct {
 	// It will be applied to Nodes allowing users run their apps on specific Node using labelSelector.
 	// required: false
 	Labels map[string]string `json:"labels,omitempty"`
+	// List of taints to set on new nodes
+	Taints []TaintSpec `json:"taints"`
```
Comment (p0lyn0mial) [inline api/pkg/api/v1/types.go]: I see that this value is not required thus can we tag it as ``json:"taints,omitempty"`` ?

Comment (mrIncompetent) [reply api/pkg/api/v1/types.go]: Done :)

**Comment by mrIncompetent** [comment]
@p0lyn0mial PTAL

**Pull request merged**